### DISCUSSION TITLE
How to open mini-browser with an URL programmatically

### DISCUSSION THREAD

**Discussion by glidingvish** [eclipse-theia/theia#12036] [Q&A] [answered]
Hi everyone,
I am trying to do following things:
1. run bash command (for example: "npm start")
2. open an URL (example: localhost:3000) in mini-browser (@theia/mini-browser extension)

I want to do it upon clicking on a menu item/widget.
following the "hello world" extension making tutorial I can run certain instructions upon menu item click. but not sure how to do the mentioned actions. specifically opening the URL in mini-browser upon clicking on menu item. I am not sure **which command to use/which argument to pass in the command** to do so.
It would be great if I can have some sample code to follow.
Thank you very much.

**Comment by msujew** [comment, marked as answer]
Hey @glidingvish,

you can simply call the `MiniBrowserCommands.OPEN_URL` command with a URL as a parameter like this:

```ts
const url = 'http://localhost:3000';
await commandRegistry.executeCommand(MiniBrowserCommands.OPEN_URL.id, url);
```

**Reply by glidingvish** [reply]
@msujew Thank you very much. worked! :)
